### PR TITLE
Added NSG to SQL MI templates as they are now supported

### PR DESCRIPTION
**Pull request by srdan-bozovic-msft**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added NSG to template to _101-sqlmi-new-vnet_
* Added NSG to template to _201-sqlmi-new-vnet-w-jumpbox_
* Added NSG to template to _201-sqlmi-new-vnet-w-point-to-site-vpn_

### CODE REVIEW

File: 101-sqlmi-new-vnet/README.md [hunk 1]
```diff
@@ -34,9 +34,7 @@ You can click the "Deploy to Azure" button at the beginning of this document or
 
 ## Important
 
-**SQL MI** is still in gated public preview. Before deploying this template you have to whitelist your subscription as explained here: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-create-tutorial-portal#whitelist-your-subscription. 
-
-During the public preview deployment might take up to 48h. The reason why provisioning takes some time is that along the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
+Usual deployment time is 3-6h but during the public preview deployment might take up to 48h. The reason why provisioning takes some time is that along the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/README.md]: Might just be a grammar issue, but I'm not sure what you're trying to say here:

The reason why provisioning takes some time is that along the Managed Instance virtual cluster that hosts the instance is created.

File: 201-sqlmi-new-vnet-w-jumpbox/README.md [hunk 2]
```diff
@@ -35,9 +35,8 @@ You can click the "Deploy to Azure" button at the beginning of this document or
 
 
 ## Important
-**SQL MI** is still in gated public preview. Before deploying this template you have to whitelist your subscription as explained here: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-create-tutorial-portal#whitelist-your-subscription. 
 
-During the public preview deployment might take up to 48h. The reason why provisioning takes some time is that the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
+Usual deployment time is 3-6h but during the public preview deployment might take up to 48h. The reason why provisioning takes some time is that the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
```
Comment (bmoore-msft) [inline 201-sqlmi-new-vnet-w-jumpbox/README.md]: same here

File: 201-sqlmi-new-vnet-w-point-to-site-vpn/README.md [hunk 3]
```diff
@@ -60,9 +60,8 @@ You can click the "Deploy to Azure" button at the beginning of this document or
 
 
 ## Important
-**SQL MI** is still in gated public preview. Before deploying this template you have to whitelist your subscription as explained here: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-create-tutorial-portal#whitelist-your-subscription. 
 
-During the public preview deployment might take up to 48h. The reason why provisioning takes some time is that the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
+Usual deployment time is 3-6h but during the public preview deployment might take up to 48h. The reason why provisioning takes some time is that the Managed Instance virtual cluster that hosts the instance is created. Each subsequent instance creation takes just about a few minutes.
```
Comment (bmoore-msft) [inline 201-sqlmi-new-vnet-w-point-to-site-vpn/README.md]: and here

File: 201-sqlmi-new-vnet-w-point-to-site-vpn/scripts/deploy.ps1 [hunk 4]
```diff
@@ -10,16 +10,52 @@ $parameters.Remove('subscriptionId')
...
+function VerifyManagedInstanceName
+{
+    param($managedInstanceName)
+    Write-Host "Verifying Managed Instance name, must be globally unique."
+    if([string]::IsNullOrEmpty($managedInstanceName))
+    {
+        Write-Host "Managed Instance name is required parameter." -ForegroundColor Red
+        break;
+    }
+    if($null -ne (Resolve-DnsName ($managedInstanceName+'.provisioning.database.windows.net') -ErrorAction SilentlyContinue))
```
Comment (bmoore-msft) [inline 201-sqlmi-new-vnet-w-point-to-site-vpn/scripts/deploy.ps1]: is this script still needed post GA?  If so, we need to update this check to not hard-code the uri

File: 101-sqlmi-new-vnet/README.md (hunk 1 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/README.md]: fixed

File: 201-sqlmi-new-vnet-w-jumpbox/README.md (hunk 2 above)
Comment (srdan-bozovic-msft) [reply 201-sqlmi-new-vnet-w-jumpbox/README.md]: fixed

File: 201-sqlmi-new-vnet-w-point-to-site-vpn/README.md (hunk 3 above)
Comment (srdan-bozovic-msft) [reply 201-sqlmi-new-vnet-w-point-to-site-vpn/README.md]: fixed

File: 201-sqlmi-new-vnet-w-point-to-site-vpn/scripts/deploy.ps1 (hunk 4 above)
Comment (srdan-bozovic-msft) [reply 201-sqlmi-new-vnet-w-point-to-site-vpn/scripts/deploy.ps1]: Post GA it will be replaced by API call, for now this is the way to check

**Comment by srdan-bozovic-msft** [comment]
@bmoore-msft updated based on feedback

**Pull request merged**